weasyl/profile.py

```diff
@@ -290,14 +290,16 @@ def select_relation(userid, otherid):
             (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (userid, otherid) = (%i, %i))),
             (SELECT EXISTS (SELECT 0 FROM frienduser WHERE userid IN (%i, %i) AND otherid IN (%i, %i) AND settings !~ 'p')),
             (SELECT EXISTS (SELECT 0 FROM ignoreuser WHERE (userid, otherid) = (%i, %i))),
-            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE (userid, otherid) = (%i, %i) AND settings ~ 'p'))
-    """, [userid, otherid, userid, otherid, userid, otherid, userid, otherid, userid, otherid], ["single"])
+            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE (userid, otherid) = (%i, %i) AND settings ~ 'p')),
+            (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (otherid, userid) = (%i, %i)))
+    """, [userid, otherid, userid, otherid, userid, otherid, userid, otherid, userid, otherid, otherid, userid], ["single"])
```

Comment:
If this is replaced with named parameters (and the query statement switched to d.engine.execute), it will be easier to read and maintain than the long list of getting userid and otherid in the right order.